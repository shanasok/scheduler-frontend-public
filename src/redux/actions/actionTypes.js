export const LOAD_COURSES = "LOAD_COURSES";
export const SET_CURRENT_LESSON = "SET_CURRENT_LESSON";
export const SET_CURRENT_PAGE = "SET_CURRENT_PAGE";
export const SET_CURRENT_LESSON_CONTENT = "SET_CURRENT_LESSON_CONTENT";
export const SET_EDIT_SLIDE_TRUE = "SET_EDIT_SLIDE_TRUE";
export const SET_EDIT_SLIDE_FALSE = "SET_EDIT_SLIDE_FALSE";
export const SET_PAGE_CONTENT_EDITABLE = "SET_PAGE_CONTENT_EDITABLE";