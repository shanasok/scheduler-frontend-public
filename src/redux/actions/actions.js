import { LOAD_COURSES } from "./actionTypes";

// Actions are payloads of information that send data from your application to your store.

export const loadCourses = courses => ({
    type: LOAD_COURSES,
    payload: {
        courses: courses
    }
});
