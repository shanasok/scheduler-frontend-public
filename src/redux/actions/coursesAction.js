import { LOAD_COURSES } from "./actionTypes";

// Actions are payloads of information that send data from your application to your store.

export function setCoursesJson(coursesJson) {
    return {
        type: LOAD_COURSES,
        payload: {
            courses: coursesJson
        }
    }
}
