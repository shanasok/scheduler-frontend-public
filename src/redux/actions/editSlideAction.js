import { SET_EDIT_SLIDE_TRUE, SET_EDIT_SLIDE_FALSE } from "./actionTypes"

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setEditSlideTrue() {
    return {
        type: SET_EDIT_SLIDE_TRUE,
        payload: {
            setEditSlide: true
        }
    }
}

export function setEditSlideFalse() {
    return {
        type: SET_EDIT_SLIDE_FALSE,
        payload: {
            setEditSlide: false
        }
    }
}