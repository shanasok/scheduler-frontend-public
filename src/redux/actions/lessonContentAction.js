import { SET_CURRENT_LESSON_CONTENT } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentLessonContent(currentLessonContent) {
    return {
        type: SET_CURRENT_LESSON_CONTENT,
        payload: {
            currentLessonContent: currentLessonContent
        }
    }
}