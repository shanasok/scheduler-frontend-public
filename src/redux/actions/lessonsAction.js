import { SET_CURRENT_LESSON } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentLesson(currentLesson) {
    return {
        type: SET_CURRENT_LESSON,
        payload: {
            currentLesson: currentLesson
        }
    }
}
