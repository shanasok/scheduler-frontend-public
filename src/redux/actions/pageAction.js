import { SET_CURRENT_PAGE } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentPage(currentPage) {
    return {
        type: SET_CURRENT_PAGE,
        payload: {
            currentPage: currentPage
        }
    }
}
