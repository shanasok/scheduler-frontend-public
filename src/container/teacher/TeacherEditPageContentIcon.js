import React from 'react';
import {bindActionCreators} from "redux";
import {connect} from "react-redux";
import {setCurrentLessonContent} from "../../redux/actions/lessonContentAction"

import '../../css/SlideStyles.css';

export function showTextBox(lessonContent, currentPage, pageContentId, setCurrentLessonContent){
    let lessonContentWithEditFlagSet = updateLessonContentToMakeItEditable(lessonContent, currentPage, pageContentId);
    setCurrentLessonContent(lessonContentWithEditFlagSet);
}

export function updateLessonContentToMakeItEditable(lessonContent, currentPage, pageContentId){
    let lessonContentCopy = JSON.parse(JSON.stringify(lessonContent));
    let page = lessonContentCopy.pages.filter(pages => pages.id === currentPage.id);
    let pageContent = page[0].pageContent.filter(pageContent => pageContent.content.id === pageContentId);
    pageContent[0].isEditable = true;
    return lessonContentCopy;
}

class TeacherEditPageContentIcon extends React.Component {

    render() {
        return <i className="fas fa-edit mouseOverBecomeArrow"
                  onClick={() => showTextBox(this.props.currentLessonContent,
                      this.props.currentPage,
                      this.props.pageContentId,
                      this.props.actions.setCurrentLessonContent)}>
        </i>
    }
}

const mapStateToProps = state => ({
    currentLessonContent: state.currentLessonContent,
    currentPage: state.currentPage,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setCurrentLessonContent
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(TeacherEditPageContentIcon);