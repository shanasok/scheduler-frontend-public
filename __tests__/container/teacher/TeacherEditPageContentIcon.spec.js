import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import {Provider as ReduxProvider} from "react-redux";
import {setCurrentLessonContent} from "../../../src/redux/actions/lessonContentAction";
import TeacherEditPageContentIcon, {updateLessonContentToMakeItEditable, showTextBox} from "../../../src/container/teacher/TeacherEditPageContentIcon";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {MemoryRouter} from "react-router";

const mockStore = configureMockStore()
const mockState = {
    slideEditMode: true,
}
let store = mockStore(mockState)

it('renders correctly when setEditSlide boolean is true', () => {

    const tree = renderer
        .create( <ReduxProvider store={store}><TeacherEditPageContentIcon/></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

const pageContent1 = {
    contentNumber: 1,
    content: {
        contentType: "FACT",
        fact: "Test Fact",
        id: 1,
        isEditable: false,
    }
};
const currentPage = {
    id: 1,
    pageNumber: 1,
    pageContent: [pageContent1],
};

const lesson = {
    id: 1,
    name: "test lesson name",
    lessonNumber: 1,
    pages: [currentPage],
};

it('correctly sets pageContent.isEditable to true', () => {
    let lessonContentCopy = updateLessonContentToMakeItEditable(lesson,currentPage,1);
    expect(lessonContentCopy.pages[0].pageContent[0].isEditable).toEqual(true);
});

it('correctly shows text box when icon is clicked', () => {
    const mockFunction = jest.fn(setCurrentLessonContent);
    showTextBox(lesson, currentPage, 1, mockFunction)
    expect(mockFunction.mock.calls.length).toBe(1);
});

configure({adapter: new Adapter()});
it('click icon should call correct action', () => {
    const mockStore = configureMockStore();
    let mockSetCurrentLessonContent = jest.fn(setCurrentLessonContent);

    const pageContent1 = {
        contentNumber: 1,
        isEditable: false,
        content: {
            contentType: "FACT",
            fact: "Test Fact",
            id: 1,
        }
    };

    const pageContent2 = {
        contentNumber: 2,
        isEditable: false,
        content: {
            contentType: "FACT",
            fact: "Test Fact",
            id: 2,
        }
    };

    const otherPage = {
        id: 2,
        pageNumber: 2,
        pageContent: [pageContent2],
    };
    const currentPage = {
        id: 1,
        pageNumber: 1,
        pageContent: [pageContent1],
    }
    const lesson = {
        id: 1,
        name: "test lesson name",
        lessonNumber: 1,
        pages: [currentPage, otherPage],
    }
    const properties = {
        pageContentId: 1,
        lessonContent: lesson,
        currentLessonContent: lesson,
        currentPage: currentPage,
        actions: { setCurrentLessonContent: mockSetCurrentLessonContent },
    };
    let store = mockStore(properties)

    let wrapper = mount(
        <MemoryRouter initialEntries={[ '/' ]}>
            <ReduxProvider store={store}>
                <TeacherEditPageContentIcon pageContentId={1}/>
            </ReduxProvider>
        </MemoryRouter>
    );


    wrapper.find('.fa-edit').at(0).simulate('click');
    expect(store.getActions()[0].type).toEqual('SET_CURRENT_LESSON_CONTENT');
});
