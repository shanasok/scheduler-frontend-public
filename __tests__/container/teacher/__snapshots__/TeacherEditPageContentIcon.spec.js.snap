// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when setEditSlide boolean is true 1`] = `
<i
  className="fas fa-edit mouseOverBecomeArrow"
  onClick={[Function]}
/>
`;
